include/rmm/mr/host/new_delete_resource.hpp

```diff
@@ -42,29 +44,21 @@ class new_delete_resource final : public host_memory_resource {
    * @param alignment Alignment of the allocation
    * @return void* Pointer to the newly allocated memory
    *---------------------------------------------------------------------------**/
-  void* do_allocate(std::size_t bytes, std::size_t alignment =
-                                           alignof(std::max_align_t)) override {
+  void *do_allocate(
+      std::size_t bytes,
+      std::size_t alignment = detail::RMM_DEFAULT_HOST_ALIGNMENT) override {
 #if __cplusplus >= 201703L
     return ::operator new(bytes, std::align_val_t(alignment));
 #else
-    // don't allocate anything if the user requested zero bytes
-    if (0 == bytes) {
-      return nullptr;
-    }
-    // allocate memory for bytes, plus potential alignment correction,
-    // plus store of the correction offset
-    void *p = ::operator new(bytes + alignment + sizeof(std::size_t));
-    std::size_t ptr_int = reinterpret_cast<std::size_t>(p);
-    // calculate the offset, i.e. how many bytes of correction were necessary
-    // to get an aligned pointer
-    std::size_t offset =
-        (ptr_int % alignment) ? (alignment - ptr_int % alignment) : 0;
-    // calculate the return pointer
-    char *ptr = static_cast<char *>(p) + offset;
-    // store the offset right after the actually returned value
-    std::size_t *offset_store = reinterpret_cast<std::size_t *>(ptr + bytes);
-    *offset_store = offset;
-    return static_cast<void *>(ptr);
+
+    // If the requested alignment isn't supported, use default
+    alignment = (detail::is_supported_alignment(alignment))
```

Comment:
Should there be an error or warning when the requested alignment isn't supported?

Reply:
This comes straight from the documentation and requirements of `std::pmr::memory_resource`: 

> Allocates storage with a size of at least bytes bytes. The returned storage is aligned to the specified alignment if such alignment is supported, and to alignof(std::max_align_t) otherwise.

This is also documented in the `host_memory_resource`.

Reply:
Seems strange to fail silently, but I suppose that's fine if documented.

Reply:
It's not failing. It's still allocating memory, but with a different alignment.